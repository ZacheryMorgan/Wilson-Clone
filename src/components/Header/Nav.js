import React from "react";
import { Link } from "react-router-dom";
import { FiUser } from "react-icons/fi";
import { BiSearch, BiShoppingBag } from "react-icons/bi";
import { GiHamburgerMenu, GiNautilusShell } from "react-icons/gi";
import "../../styles/Nav.scss";

const Nav = ({ onClick }) => {
  return (
    <nav className="border-b">
      <div className="nav-wrapper flex justify-between items-center h-12 px-5 max-w-screen-2xl mx-auto xl:h-20">
        <Link to="/">
          <GiNautilusShell className="w-8 h-8 mr-10 xl:w-12 xl:h-12 min-w-min  hover:cursor-pointer" />
        </Link>
        <ul className="site-links flex gap-14">
          <Link to="/shop">
            <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
              Shells
            </li>
          </Link>
          <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
            Equipment
          </li>
          <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
            Custom
          </li>
        </ul>
        <ul className="flex gap-1 nav-links">
          <button className="">
            <BiSearch className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6" />
          </button>
          <button className="hidden xl:block">
            <FiUser className="hover:stroke-red-600 cursor-pointer transition-all duration-300 w-6 h-6" />
          </button>
          <button className="" onClick={onClick}>
            <BiShoppingBag className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 " />
          </button>
          <button className="xl:hidden">
            <GiHamburgerMenu className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 active:border-black after:border-2" />
          </button>
        </ul>
      </div>
    </nav>
  );
};
export default Nav;
