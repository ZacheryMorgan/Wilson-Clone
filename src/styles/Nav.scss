.site-links {
  li {
    &:hover {
      &::after {
        content: "";
        display: block;
        position: absolute;
        bottom: 0.1rem;
        left: 0;
        width: 100%;
        height: 3px;
        background: #101820;
      }
    }
  }
}

.nav-links {
  button {
    padding: 7px;

    &:focus {
      outline: -webkit-focus-ring-color auto 5px;
    }
  }
}