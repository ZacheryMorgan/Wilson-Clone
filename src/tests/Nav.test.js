import React from "react";
import { render } from "@testing-library/react";
import Nav from "../components/Nav";

describe("Nav component", () => {
  it("Renders Nav component correctly", () => {
    const { container } = render(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
