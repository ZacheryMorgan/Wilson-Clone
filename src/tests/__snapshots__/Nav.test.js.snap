// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component Renders Nav component correctly 1`] = `
<div>
  <nav
    class="border-b"
  >
    <div
      class="nav-wrapper flex justify-between items-center h-12 px-5 max-w-screen-2xl mx-auto xl:h-20"
    >
      <svg
        class="w-8 h-8 mr-10 xl:w-12 xl:h-12 min-w-min  hover:cursor-pointer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M141.49 361.44c10.6 2.434 16.243 2.523 27.81 1.153-8.665 41.136 46.103 47.286-16.42 118.757 21.354-80.17-25.916-60.54-11.39-119.91zm-40.532-20.274l25.917 13.68c-16.114 18.798-4.805 80.966-38.226 112.614-1.29-49.98-4.846-91.22 12.308-126.294zm264.044 115.452l1.547-33.994c-12.31-35.97-29.496-68.375-48.6-99.386 11.67-4.796 21.683-11.94 28.495-23.506 32.502 10.19 65.77 18.718 103.352 17.21 8.706 62.95-27.626 109.003-84.795 139.676zm-177.17-75.106l21.3-6.7c26.714-18.29 48.23-36.024 64.4-58.176 8.125 4.756 16.735 9.238 26.6 12.712 18.042 34.365 32.07 67.214 40.21 102.514-41.305 30.874-118.877 26.916-152.51-50.35zM68.98 315.588l21.233 23.216c-9.738 13.462-20.274 38.548-30.584 80.837-14.157-39.136-10.908-74.073 9.35-104.052zm-28.392-26.61l16.34 20.782c-8.996 10.657-19.41 30.97-30.697 65.73-9.398-33.655-4.456-62.724 14.358-86.512zm213.58-57.984c-2.62 9.73-3.756 19.307-1.724 28.343l9.61 41.426c-16.85 20.87-37.88 39.935-66.4 56.695-15.35-54.703 15.63-97.823 58.515-126.464zM68.82 296.95c-29.31-26.585-31.01-42.933-15.51-52.8.492-66.255 29.545-109.35 73.365-139.152l-4.28 43.345c3.748 43.708 23.925 80.24 57.62 133.243-9.31 21.354-14.227 45.24-8.23 66.988-38.17-10.616-81.022-29.278-102.965-51.624zm236.99 4.136l11.472-47.02c33.01 14.856 17.823 38.483-11.472 47.02zm183.787-.234l-33.188-11.6c-36.356 2.137-70.495-2.676-103.787-10.407 1.733-14.494-1.58-29.205-7.2-42.506 15.02-26.33 50.117-64.756 78.855-87.916 49.447 49.27 70.69 100.152 65.32 152.428zm-199.312-1.144l-8.85-4.257c-5.7-16.177-10.094-33.315-12.44-51.96 12.358-5.894 23.28-7.167 32.55-3.177l-11.26 59.395zM143.44 134.656c34.454-64.264 82.805-93.792 135.34-104.006l-20.323 42.547c-5.788 41.636-8.472 85.255 1.363 136.798-22.58 10.1-47.158 22.37-67.964 53.768-25.7-38.178-46.247-81.12-48.415-129.107zm136.96 75.96c-9.254-41.2-6.32-83.707-.63-135.815 67.086-10.962 125.078 3.436 167.4 43.137-13.842 3.007-26.305 7.892-38.163 13.64-30.52 28.012-55.25 60.264-77.217 94.468-15.26-8.682-32.672-13.333-51.39-15.43z"
        />
      </svg>
      <ul
        class="site-links flex gap-14"
      >
        <li
          class="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block"
        >
          Sportswear
        </li>
        <li
          class="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block"
        >
          Equipment
        </li>
        <li
          class="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block"
        >
          Custom
        </li>
      </ul>
      <ul
        class="flex gap-1 nav-links"
      >
        <button
          class=""
        >
          <svg
            class="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </button>
        <button
          class="hidden xl:block"
        >
          <svg
            class="hover:stroke-red-600 cursor-pointer transition-all duration-300 w-6 h-6"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
        </button>
        <button
          class=""
        >
          <svg
            class="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 22h14c1.103 0 2-.897 2-2V9a1 1 0 0 0-1-1h-3V7c0-2.757-2.243-5-5-5S7 4.243 7 7v1H4a1 1 0 0 0-1 1v11c0 1.103.897 2 2 2zM9 7c0-1.654 1.346-3 3-3s3 1.346 3 3v1H9V7zm-4 3h2v2h2v-2h6v2h2v-2h2l.002 10H5V10z"
            />
          </svg>
        </button>
        <button
          class="xl:hidden"
        >
          <svg
            class="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 active:border-black after:border-2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
            />
          </svg>
        </button>
      </ul>
    </div>
  </nav>
</div>
`;
